docker/base.Dockerfile

```diff
@@ -1,33 +1,16 @@
-FROM ubuntu:20.04
+ARG PYTHON_VERSION=3.9
+FROM python:${PYTHON_VERSION}-slim AS base
 
-# python
 ENV PYTHONFAULTHANDLER=1 \
     PYTHONUNBUFFERED=1 \
     PYTHONHASHSEED=random \
-    PIP_NO_CACHE_DIR=off \
-    PIP_DISABLE_PIP_VERSION_CHECK=on
-
-RUN apt-get update && \
-  apt-get install --no-install-recommends -q -y \
-  build-essential \
-  ca-certificates \
-  libsnappy-dev \
-  protobuf-compiler \
-  libprotobuf-dev \
-  python3 \
-  python3-dev \
-  python-is-python3 \
-  python3-venv \
-  python3-pip \
-  curl \
-  unzip \
-  git && \
-  apt-get autoclean && \
-  apt-get autoremove --purge
-
-RUN curl -sSL https://bootstrap.pypa.io/get-pip.py | python && \
-  pip install --no-cache-dir --upgrade --pre pip
+    PIP_NO_CACHE_DIR=1 \
+    PIP_DISABLE_PIP_VERSION_CHECK=1
 
 ARG ZENML_VERSION
+
 # install the given zenml version (default to latest)
-RUN pip install --no-cache-dir zenml${ZENML_VERSION:+==$ZENML_VERSION}
+RUN pip install zenml${ZENML_VERSION:+==$ZENML_VERSION}
```

Comment:
I thought using `--no-cache-dir` in your dockerfile results in a smaller image. Was that not the case ?

That being said, caching may speed up the process of building the image, but in that case you might want to [remove the cache](https://pip.pypa.io/en/stable/topics/caching/#removing-the-cache) at the end of the build.

Reply:
I ran some tests and the environment variable `PIP_NO_CACHE_DIR` set at the top seemed to be enough

Reply:
Ah, I missed that :) good work !